ESP8266Interface.cpp

```diff
@@ -43,8 +43,11 @@ ESP8266Interface::ESP8266Interface(PinName tx, PinName rx, bool debug)
     memset(_cbs, 0, sizeof(_cbs));
     memset(ap_ssid, 0, sizeof(ap_ssid));
     memset(ap_pass, 0, sizeof(ap_pass));
+    ap_sec = NSAPI_SECURITY_UNKNOWN;
 
     _esp.attach(this, &ESP8266Interface::event);
+    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
```

Comment:
This puzzles me - my impression of the code was that the timeout was generally left at MISC, and this was assumed on call entry? I guess it was previously unset by construction, and hence at ATCmdParser default?

Reply:
Earlier when reset was called from connect() this was set in a similar manner.

Reply:
But the timeout was being set up in preparation for the connect commands then, not the reset, right?